File: Class_member/2.rb
class Cs
    def Cs.class_method() #클래스 메소드를 사용 할려면 클래스 명을 써야함.
        p("Class method")
    end
    def instance_method() # 인스턴스의 소속으로 사용
        p("Instance method")
    end
end

i = Cs.new()
Cs.class_method()
i.instance_method()

# i.class_method() 에러가 뜬다. 
# Cs.instance_method() 에러가 뜬다.

File: Class_member/2.py
class Cs :
    @staticmethod
    def static_method() :
        print("static method")
    @classmethod
    def class_method(cls) : # self처럼 cls 를 써줘야함.
        print("class method")
    def instance_method(self) : # 인스턴스는 self로 첫 번째 인자를 줘야함
        print("instance method")

i = Cs()
#클래스 멤버 메소드
Cs.static_method() 
Cs.class_method()

i.instance_method()